content/blog/2025-03-13-cve-2025-27363-patches.md

```diff
@@ -0,0 +1,86 @@
+
+From the announcement: 
+> Description
+> An out of bounds write exists in FreeType versions 2.13.0 and below (newer versions of FreeType are not vulnerable) when attempting to parse font subglyph structures related to TrueType GX and variable font files. The vulnerable code assigns a signed short value to an unsigned long and then adds a static value causing it to wrap around and allocate too small of a heap buffer. The code then writes up to 6 signed long integers out of bounds relative to this buffer. This may result in arbitrary code execution. This vulnerability may have been exploited in the wild.
+
+More information about this vulnerability can be found at these links:
+
+* <https://nvd.nist.gov/vuln/detail/CVE-2025-27363>  
+* <https://www.facebook.com/security/advisories/cve-2025-27363>  
+* <https://access.redhat.com/security/cve/cve-2025-27363>  
+* <https://seclists.org/oss-sec/2025/q1/207>
+* <https://thehackernews.com/2025/03/meta-warns-of-freetype-vulnerability.html>  
+
+# Impact and Mitigation:
+
+While this flaw is not yet known to have been exploited in the wild, it's devious and highly-enough rated that we are pulling in the patches for testing ahead of our upstream. Thes patches have been tested, but require further testing before we will release them to production. To help us out, please install the testing repo and then the updated packages and please share the results as soon as you can, either in [AlmaLinux chat](https://chat.almalinux.org), on [bugs.almalinux.org](https://bugs.almalinux.org), or by emailing [packager@almalinux.org](packager@almalinux.org).
+
+
+## Installing the testing repo on AlmaLinux
+
+It only takes a few steps to install and test the patched version of OpenSSH in the testing repo. 
+
+**Install the testing repo**
+
+```bash
+dnf install -y almalinux-release-testing
+``` 
+
+**Then update the package:**
+
+```bash
+dnf update freetype
+``` 
+
+**Confirm you have the patched version of the package** 
+
+The patched RPM versions are listed further down in this article.
+
+```bash
+rpm -qa freetype
+``` 
+
+Note: We don't recommend that you keep the testing repo enabled after you've updated the package, unless you've done this on a truly non-production environment. If this is a production environment, you can disable the repo with this command:
+
+
+```bash
+dnf config-manager --disable almalinux-testing
+```
+
+If you encounter problems, please let us know as soon as you can, either in [AlmaLinux chat](https://chat.almalinux.org), on [bugs.almalinux.org](https://bugs.almalinux.org), or by emailing [packager@almalinux.org](packager@almalinux.org).
+
+## AlmaLinux OS 8 & AlmaLinux OS  9
+
+Both AlmaLinux 8 and AlmaLinux 9 are vulnerable to this CVE. Patches are not yet available from RedHat, so the [patch we are pulling in](https://git.almalinux.org/rpms/freetype/pulls/1) is [from FreeType](https://gitlab.freedesktop.org/freetype/freetype/-/commit/ef636696524b081f1b8819eb0c6a0b932d35757d) itself.  
```

Comment:
This may change more.  Honestly don't merge this yet.

```suggestion
Both AlmaLinux 8 and AlmaLinux 9 are vulnerable to this CVE. Patches are not yet available from RedHat, so the [patch we are pulling in](https://git.almalinux.org/rpms/freetype/pulls/2) is [from FreeType](https://gitlab.freedesktop.org/freetype/freetype/-/commit/ef636696524b081f1b8819eb0c6a0b932d35757d) itself.

We'd like to thank Michel Lind of Meta for the backporting work in his [PR to CentOS Stream 9](https://gitlab.com/redhat/centos-stream/rpms/freetype/-/merge_requests/8) as well as for bringing this vulnerability to our attention.
```